modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxAdapter.java

```diff
@@ -275,13 +275,15 @@
     protected ConsistentIdMapper consistentIdMapper;
 
     /** Mvcc tx update snapshot. */
+    @GridToStringInclude
     protected volatile MvccSnapshot mvccSnapshot;
 
     /** Rollback finish future. */
     @GridToStringExclude
     private volatile IgniteInternalFuture rollbackFut;
 
     /** */
+    @SuppressWarnings("unused")
```

Comment:
`@GridToStringExclude`?

Reply:
No. This field is changed by AtomicReferenceFieldUpdater, so IDEA thinks it is unused.

Reply:
OK, I get it. I'll fix it in next patch, I think. May be it would be better to override `toString` method in txCounters class.